lib/common/xml.c

```diff
@@ -786,6 +787,64 @@ pcmk__xe_set_content(xmlNode *node, const char *format, ...)
     }
 }
 
+/*!
+ * \internal
+ * \brief Check whether the first character of a string is an XML NameStartChar
+ *
+ * See https://www.w3.org/TR/xml/#NT-NameStartChar.
+ *
+ * This is almost identical to libxml2's \c xmlIsDocNameStartChar(), but they
+ * don't expose it as part of the public API.
+ *
+ * \param[in]  utf8  UTF-8 encoded string
+ * \param[out] len   Where to store size in bytes of first character in \p utf8
+ *
+ * \return \c true if \p utf8 begins with a valid XML NameStartChar, or \c false
+ *         otherwise
+ */
+bool
+pcmk__xml_is_name_start_char(const char *utf8, int *len)
+{
+    int c = 0;
+
+    /* xmlGetUTF8Char() abuses the len argument. At call time, it must be set to
+     * "the minimum number of bytes present in the sequence... to assure the
+     * next character is completely contained within the sequence." It's similar
+     * to the "n" in the strn*() functions. However, this doesn't make any sense
+     * for null-terminated strings, and there's no value that indicates "keep
+     * going until '\0'." So we set it to 4, the max number of bytes in a UTF-8
+     * character.
+     *
+     * At return, it's set to the actual number of bytes in the char, or 0 on
+     * error.
+     */
+    *len = 4;
+
+    // Note: xmlGetUTF8Char() assumes a 32-bit int
+    c = xmlGetUTF8Char((pcmkXmlStr) utf8, len);
+    if (c < 0) {
+        crm_err("Invalid UTF-8 character 0x%X", c);
```

Comment:
I'm thinking info or debug. The caller would have the context to know whether it's a problem or not. If you think the distinction is important, you could return a standard code instead of bool